lib/AST/ASTContext.cpp

```diff
@@ -2803,16 +2803,19 @@ ProtocolCompositionType::build(const ASTContext &C, ArrayRef<Type> Protocols) {
     return Result;
 
   bool isCanonical = true;
+  RecursiveTypeProperties properties;
   for (Type t : Protocols) {
     if (!t->isCanonical())
       isCanonical = false;
+    properties |= t->getRecursiveProperties();
   }
 
   // Create a new protocol composition type.
   ProtocolCompositionType *New
     = new (C, AllocationArena::Permanent)
```

Comment:
I just wanted to give @jrose-apple another fun memory smasher bug to track down :)